cmd/frontend/internal/httpapi/graphql.go

```diff
@@ -134,12 +159,30 @@ func serveGraphQL(logger log.Logger, schema *graphql.Schema, rlw graphqlbackend.
 			costHistogram.WithLabelValues(actorTypeLabel(isInternal, anonymous, requestSource)).Observe(float64(cost.FieldCount))
 
 			rl := conf.RateLimits()
-			if !isInternal && (exceedsLimit(cost.AliasCount, rl.GraphQLMaxAliases) ||
-				exceedsLimit(cost.HighestDuplicateFieldCount, rl.GraphQLMaxDuplicateFieldCount) ||
-				exceedsLimit(cost.UniqueFieldCount, rl.GraphQLMaxUniqueFieldCount) ||
-				exceedsLimit(cost.FieldCount, rl.GraphQLMaxFieldCount)) {
-				writeViolationError(w)
-				return
+			limits := []struct {
+				cost          int
+				limit         int
+				violationType string
+			}{
+				{cost.AliasCount, rl.GraphQLMaxAliases, "alias count"},
+				{cost.FieldCount, rl.GraphQLMaxFieldCount, "field count"},
+				{cost.MaxDepth, rl.GraphQLMaxDepth, "query depth"},
+				{cost.HighestDuplicateFieldCount, rl.GraphQLMaxDuplicateFieldCount, "duplicate field count"},
+				{cost.UniqueFieldCount, rl.GraphQLMaxUniqueFieldCount, "unique field count"},
+			}
+
+			if !isInternal {
```

Comment:
This would group the logic a bit nicer in one block 

```suggestion
			if !isInternal {
    			limits := []struct {
				cost          int
				limit         int
				violationType string
			}{
				{cost.AliasCount, rl.GraphQLMaxAliases, "alias count"},
				{cost.FieldCount, rl.GraphQLMaxFieldCount, "field count"},
				{cost.MaxDepth, rl.GraphQLMaxDepth, "query depth"},
				{cost.HighestDuplicateFieldCount, rl.GraphQLMaxDuplicateFieldCount, "duplicate field count"},
				{cost.UniqueFieldCount, rl.GraphQLMaxUniqueFieldCount, "unique field count"},
			}
```

Reply:
Ah, I missed that. Thanks!